### DISCUSSION TITLE
BTT UPS 24V and power loss failure recovery

### DISCUSSION THREAD

**Discussion by patofoto** [Jyers/Marlin#1502] [General Discussion]
I am working on installing my BTT UPS 24V V1.0 on my Ender3 v2 to enable power loss failure recovery. I believe I have to plug the BTT UPS to the filament sensor port on the Creality board:

![Creality-3D-Ender-3-V2-Mainboard-32-bit--2002040059-25618_2](https://user-images.githubusercontent.com/37857628/138129814-e87b522a-dc3a-4316-8a09-a61681195c2a.png)

When enabling Power Loss Recovery in the Marlin firmware, I need to specify a POWER_LOSS_PIN number:

<img width="1013" alt="Screen Shot 2021-10-20 at 5 32 59 AM" src="https://user-images.githubusercontent.com/37857628/138129891-a98b7a33-f2cf-4cf9-9cc9-1f7bf025bc85.png">

Don't know what PIN number to use. Anybody has installed this BTT UPS on a Creality Ender3 v2 and can tell me what the PIN number, please?. Having a hard time finding any info on this. Am I wrong in thinking I should plug the UPS board into the filament sensor port on my board?.

**Comment by Cal-LF** [comment]
Can't help you on this setup. Hope you have success with the install.

**Reply by patofoto** [reply]
Thanks

**Comment by patofoto** [comment]
Sorry but not successful. Not sure what to try next.

**Comment by patofoto** [comment]
I am thinking. A lot of people use POWER_LOSS_RECOVERY without a PSU. Like this:
<img width="946" alt="Screen Shot 2021-10-22 at 1 26 32 PM" src="https://user-images.githubusercontent.com/37857628/138497729-6331088d-5bce-4436-9534-fef0445e8154.png">

A file gets written into the SD card that gets read after the power comes back. I am not getting this file at all. Even when I disable BACKUP_POWER_SUPPLY. 

Power Loss Recovery is definitely enabled:

![IMG_6152](https://user-images.githubusercontent.com/37857628/138498169-dcaaf73f-764d-48b7-8564-fd58a241345e.jpg)

Could this issue be pointing to something else I am doing wrong?.  SDSUPPORT is enabled in configuration.h

**Comment by patofoto** [comment]
Wow!!. A little more complicated than I expected. Will contact a friend of mine that knows electronics to help me. I know enough to get in trouble but not enough to know what this means. Thank you so much. Was also trying to contact BTT online but have not gotten a response. I have the Creality 4.2.2 board. My firmware is setup as you indicate. Will modify the connection and report back as soon as I can. Thank you again.

**Comment by patofoto** [comment]
For the 47uF capacitor, does voltage matter?. I am finding 16V, 50V, 100V, 400V versions. Assuming 16V?. This is a picture of how a 50V version looks:



![AEPD_1_2019073159263794](https://user-images.githubusercontent.com/37857628/138504690-a88f56fd-9777-4ce2-ab43-b0a9d60acfca.jpg)


I know you probably know this.

**Reply by patofoto** [reply]
Thank you. The only version I was able to find in my local electronics store was 25V. I realize now they have polarity. Assuming it matters which way I connect it?. Negative to ground?.

**Reply by adam3654** [reply]
Negative to ground is correct

**Comment by patofoto** [comment]
It is working now!!!. Nothing to do with the cable and noise. I am still using a single cable to connect to the signal pin. No capacitor or resistor needed. At least in my printer. The culprit of my problem was that I did not know that, for power loss failure recovery to work, you need to be printing from the SD card. My setup has OctoPrint managing my printer. Since the G-Code was coming from my OctoPrint Pi, I guess that nothing gets written to the SD card when a power failure happens. When I tried printing from the SD card, everything worked as it should. Thank you all for your help. Will continue investigating if there is a way to have OctoPrint drive my printer and still have power failure recovery work.

**Reply by patofoto** [reply]
Sorry to keep asking you stuff. I did not find the line:

#define SDIO_SUPPORT

in configuration.h

Do I write it in?. If so, where?

**Comment by patofoto** [comment]
@tititopher68-dev Thank you so much for all your help. Glad to see @Jyers is back. Working to update to fresh firmware. Hitting some compile errors. Just posted on Issues.
